packages/serial/src/ZWaveSocket.ts

```diff
@@ -17,8 +21,8 @@ export class ZWaveSocket extends ZWaveSerialPortBase {
 				create: () => new net.Socket(),
 				open: (serial: net.Socket) =>
 					new Promise((resolve) => {
-						serial.on("close", () => {
-							if (this.isOpen) {
+						serial.on("close", (err: DisconnectError) => {
+							if (err?.disconnected === true) {
```

Comment:
Are you sure about this? The type declarations claim that the parameter for the close event is a boolean. And the serialport code throws an error, so I would expect this to come through the `"error"` event.

Reply:
Ok, seems like the type declarations are incorrect:
https://serialport.io/docs/api-stream#close